fhir-tools/src/main/java/com/ibm/fhir/tools/CodeGenerator.java

```diff
@@ -306,17 +306,60 @@ private void generateAbstractVisitorClass(String basePath) {
         cb._import("java.time.Year");
         cb._import("java.time.YearMonth");
         cb._import("java.time.ZonedDateTime");
+        cb._import("javax.annotation.Generated");
         cb.newLine();
         
         cb._import("com.ibm.fhir.model.resource.*");
         cb._import("com.ibm.fhir.model.type.*");
-        cb._import("com.ibm.fhir.model.type.code.*");
         cb._import("com.ibm.fhir.model.type.Boolean");
         cb._import("com.ibm.fhir.model.type.Integer");
         cb._import("com.ibm.fhir.model.type.String");
         cb.newLine();
         
-        cb._class(mods("public", "abstract"), "AbstractVisitor", null, implementsInterfaces("Visitor"));
+        cb.javadocStart();
+        cb.javadoc("DefaultVisitor provides a default implementation of the Visitor interface which uses the");
+        cb.javadoc("value of the passed {@code visitChildren} boolean to control whether or not to");
+        cb.javadoc("visit the children of the Resource or Element being visited.");
+        cb.javadoc("");
+        cb.javadoc("Subclasses can override the default behavior in a number of places, including:");
+        cb.javadoc("<ul>",false);
+        cb.javadoc("<li>preVisit methods to control whether a given Resource or Element gets visited",false);
+        cb.javadoc("<li>visitStart methods to provide setup behavior prior to the visit",false);
+        cb.javadoc("<li>defaultAction methods to perform some common action on all visited Resources and Elements",false);
+        cb.javadoc("<li>specific visit methods to perform unique behavior that varies by the type being visited",false);
+        cb.javadoc("<li>visitEnd methods to provide initial cleanup behavior after a Resource or Element has been visited",false);
```

Comment:
should the li tags be closed?

Reply:
its proper html, but not proper xhtml.  it is consistent with our use of unclosed <p> tag in the model javadoc.  i don't feel too strongly, but when john and I discussed that a while ago we sort of agreed that the optional closing tags don't add a whole lot.

Reply:
just an observation :)

Reply:
my mind goes to balance, and I wanted to double check